src/Repeater/DefaultRepeater.ts

```diff
@@ -0,0 +1,147 @@
+import { logger } from '../Utils';
+import {
+  Repeater,
+  RepeaterEventHandler,
+  RepeaterDeployedEvent,
+  RepeaterEvents
+} from './Repeater';
+import { inject, injectable } from 'tsyringe';
+import io, { Socket } from 'socket.io-client';
+import parser from 'socket.io-msgpack-parser';
+import { hostname } from 'os';
+import { once } from 'events';
+
+export interface DefaultRepeaterOptions {
+  readonly uri: string;
+  readonly token: string;
+}
+
+export const DefaultRepeaterOptions: unique symbol = Symbol(
+  'DefaultRepeaterOptions'
+);
+
+@injectable()
+export class DefaultRepeater implements Repeater {
+  private latestReconnectionError?: Error;
+  private readonly DEFAULT_RECONNECT_TIMES = 3;
+  private socket?: Socket;
+
+  constructor(
+    @inject(DefaultRepeaterOptions)
+    private readonly options: DefaultRepeaterOptions
+  ) {}
+
+  public disconnect() {
+    if (!this.socket) {
+      throw new Error('Not connected');
+    }
+
+    this.socket?.disconnect();
+    this.socket?.removeAllListeners();
+    this.socket = undefined;
+  }
+
+  public async deploy(
+    repeaterId?: string
+  ): Promise<RepeaterDeployedEvent['response']> {
+    if (!this.socket) {
+      throw new Error('Repeater is not connected yet');
+    }
+
+    this.socket.emit('deploy', {
+      repeaterId
+    });
+
+    const [result]: RepeaterDeployedEvent['response'][] = await once(
+      this.socket,
+      'deployed'
+    );
+
+    return result;
+  }
+
+  public connect() {
+    this.socket = io(this.options.uri, {
+      parser,
+      path: '/api/ws/v1',
+      transports: ['websocket'],
+      reconnectionAttempts: this.DEFAULT_RECONNECT_TIMES,
+      auth: {
+        token: this.options.token,
+        domain: hostname()
+      }
+    });
+
+    this.socket.on('connect_error', (error: Error) => {
+      logger.debug(`Unexpected error: %s`, error);
```

Comment:
The log message must be clear and concise providing all required info:
```suggestion
      logger.debug(`Unable to connect to the %s host`, this.options.uri, error);
```